### DISCUSSION TITLE
Switch to poetry for dependency management

### DISCUSSION THREAD

**Discussion by empicano** [empicano/aiomqtt#158] [General]
What do you think about switching to [poetry](https://github.com/python-poetry/poetry) for dependency management?

I use and like it a lot for my projects, I have never used it to package something though. What I especially miss from our pip workflow is pinning dependencies. For something like our docs (in progress!) this would be quite handy.

There are not a lot of packages using it at the moment, I'd guess it's because poetry is rather new still. Or maybe you know of any downsides/quirks I'm not seeing? [Poetry itself](https://github.com/python-poetry/poetry) is obviously using it, and there's [pendulum](https://github.com/sdispater/pendulum) as well.

**Comment by JonathanPlasse** [comment]
This is what I use for my private projects.
I like it.

**Comment by JonathanPlasse** [comment]
An [example](https://github.com/requests-cache/requests-cache/blob/main/.github/workflows/deploy.yml).

**Comment by frederikaalund** [comment]
I think that is an excellent idea. 👍 I too use poetry for my other projects.
